### ISSUE TITLE
dockerfiles/atom doesn't seem to work as-is

### ISSUE THREAD

**Issue by aquacode** [jessfraz/dockerfiles#288]
First, I pulled down the image from jess/atom. This did not run for me straightaway using the run command documented in the Dockerfile.

Then, researched a bit, found in Issues that someone else had similar problem on a different image, and went ahead and added the shared library: libX11-xcb-dev. Build the Docker image, and the first error went away, but then I got the error: Gtk cannot open display.

Next, I found in a different Issue, someone also had this problem, and the only apparent workaround that seemed viable was to run: xhost local:root. Bam, that worked! But why?

Even so, upon running the image having executed `xhost local:root`, the log output showed that there were still many errors, most seemed related to UI, e.g. failed to launch GPU process, but also one related to this: canberra-gtk-module.

Found another Issue here that someone found in 'firefox' image where the aforementioned was missing. So I add that library, and that error goes away when starting up the container. But, "Failed to launch GPU process" still remains, as well as many others such as:

Failed to connect to the bus: Failed to connect to the socket: /var/run/dbus/system_bus_socket
Failed to load libGL.so 

Interestingly, Atom does open up and it is semi-usable. But alas, after trying different things it does crash.

I can probably try to add the missing shared library to the built image, libGL, but what gives? Kind of wondering how this ever worked? Or, better yet, what am I doing so wrong that I have to add these additional libraries AND execute `xhost local:root`, and still get errors?

**Comment by jessfraz** [comment]
It seems you are frustrated. A lot of these desktop apps are reliant on a
very specific setup. If you understand X and xauth the xhost: local makes
sense. You must authenticate with your X server to be able to connect to
the socket and display. The GL errors can be solved by mounting /dev/dri
(your render device) into the container with your specific graphics drivers
installed.

You should know running desktop apps in containers is different than say
nginx. A lot is dependent on the host setup and your specific graphics
settings.

**Comment by jessfraz** [comment]
You can always find missing libraries as well with ldd and always open a PR
if you find something that should be added. I personally don't use atom so
it probably not up to date.

**Comment by aquacode** [comment]
Yeah, sure, I can make the necessary updates via PR, but my confusion was around getting the impression that one should be able to simply grab the image, or build from the Dockerfile, and get up and running with minimal adjustments. So, I opened the Issue to see if there was more documentation I might be missing regarding any "host setup" or "specific graphic settings" that might not have been documented initially. 

It seems to be the case that changes I made are required, but that there are other host-specific settings that need to be discovered and adjusted in order to get these images to work (not a part of the Dockerfile or resulting image).

If there are any additional setups or settings to be made I appreciate any ideas.

FWIW, I'm not particularly interested in Atom, it just happened to be the first one I grabbed after discovering and reading your blog post, and I thought I'd try an image out and see how it goes. I then found it wasn't so straight forward.

Thanks
